### PR TITLE
Import support for provider-datadog

### PR DESCRIPTION
**Pull request by yfronto**
* Supplies definition for `ResourceImporter` so that a datadog resource can be imported via `terraform import`.

### CODE REVIEW

**Comment by stack72** [comment]
Hi @yfronto 

Thanks for the PR here - I'd love to see a test added to this functionality along the lines of https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/import_aws_elb_test.go

This has helped us identify areas of the schema that were not set to state correctly

Thanks

Paul

**Comment by yfronto** [comment]
Import tests added.

File: builtin/providers/datadog/import_datadog_monitor_test.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package datadog
+
+import (
+	"github.com/hashicorp/terraform/helper/resource"
+	"testing"
+)
+
+func TestDatadogMonitor_import(t *testing.T) {
+	resourceName := "datadog_monitor.bar"
```
Comment (stack72) [inline builtin/providers/datadog/import_datadog_monitor_test.go]: The resource you are importing is `datadog_monitor.foo` :)

**Comment by stack72** [comment]
just ran the tests:

```
% make testacc TEST=./builtin/providers/datadog TESTARGS='-run=Test'
==> Checking that code complies with gofmt requirements...
/Users/stacko/Code/go/bin/stringer
go generate $(go list ./... | grep -v /terraform/vendor/)
2016/08/19 18:22:05 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/datadog -v -run=Test -timeout 120m
=== RUN   TestDatadogMonitor_import
--- FAIL: TestDatadogMonitor_import (85.55s)
	testing.go:265: Step 1 error: Resource specified by ResourceName couldn't be found: datadog_monitor.bar
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestProvider_impl
--- PASS: TestProvider_impl (0.00s)
=== RUN   TestAccDatadogMonitor_Basic
--- PASS: TestAccDatadogMonitor_Basic (82.72s)
=== RUN   TestAccDatadogMonitor_Updated
--- PASS: TestAccDatadogMonitor_Updated (83.72s)
=== RUN   TestAccDatadogMonitor_TrimWhitespace
--- PASS: TestAccDatadogMonitor_TrimWhitespace (65.36s)
=== RUN   TestAccDatadogTimeboard_update
--- PASS: TestAccDatadogTimeboard_update (82.49s)
FAIL
exit status 1
FAIL	github.com/hashicorp/terraform/builtin/providers/datadog	399.854s
make: *** [testacc] Error 1
```

**Comment by yfronto** [comment]
Thanks, I wasn't running the tests correctly. I'll update when I have them passing.

Cheers,
-Kurt

**Comment by yfronto** [comment]
Not ready yet; tests are still failing when floats have trailing zeros.

**Comment by yfronto** [comment]
Should be set. PTAL when you get a chance.

Cheers,
-Kurt

**Comment by jen20** [comment]
Thanks @yfronto - this LGTM now!

```
› envchain datadog.terraform make testacc TEST=./builtin/providers/datadog
==> Checking that code complies with gofmt requirements...
/Users/James/Code/go/bin/stringer
go generate $(go list ./... | grep -v /terraform/vendor/)
2016/08/22 05:46:56 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/datadog -v  -timeout 120m
=== RUN   TestDatadogMonitor_import
--- PASS: TestDatadogMonitor_import (86.77s)
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestProvider_impl
--- PASS: TestProvider_impl (0.00s)
=== RUN   TestAccDatadogMonitor_Basic
--- PASS: TestAccDatadogMonitor_Basic (79.37s)
=== RUN   TestAccDatadogMonitor_Updated
--- PASS: TestAccDatadogMonitor_Updated (80.80s)
=== RUN   TestAccDatadogMonitor_TrimWhitespace
--- PASS: TestAccDatadogMonitor_TrimWhitespace (62.81s)
=== RUN   TestAccDatadogTimeboard_update
--- PASS: TestAccDatadogTimeboard_update (80.12s)
PASS
ok     	github.com/hashicorp/terraform/builtin/providers/datadog       	389.870s
```

**Comment by glasser** [comment]
I don't use the importer, but after this change I'm seeing lots of spurious diffs on `plan`.

So for example when I set a threshold as 

```
      critical = "15"
```

I get a diff of

```
    thresholds.critical: "15.0" => "15"
```

Should I be not using quotes when I specify values, maybe?

**Comment by stack72** [comment]
Hi @glasser 

I believe this has been fixed in the latest release?

P.

**Comment by glasser** [comment]
Not sure — obsessively normalized all my thresholds a while ago.

**Pull request merged**